delocate/tests/test_delocating.py

```diff
@@ -507,3 +507,150 @@ def test_bads_report():
                   set([(LIB64A, LIBBOTH, ARCH_32),
                        (LIB64, LIB64, ARCH_32, ARCH_64),
                        (LIB32, ARCH_32)]))
+
+
+def test_ld_library_path_lookups():
```

Comment:
I'm puzzled - aren't you ignoring LD_LIBRARY_PATH?

Reply:
Yeah, I didn't follow up with a comment in the main thread bringing the PR to your attention yet because I'm going through and re-verifying everything, as it's definitely a bit stale for me (sorry!). I'll put in a script you can run yourself to verify the way things work.

Reply:
Ok, so this ended up turning into two different features: 

Originally this PR was about just searching for additional libraries using environment variables, and using those to resolve `@rpath` entries. The choice of environment variables was somewhat arbitrary at the time, and I suggested using all three of the choices. That's where this test came from.

After that, you suggested, quite reasonably, that we should extend library resolution more generally to using environment variables, and I tried to do something a bit more formal like Apple has defined in their developer documentation. This has turned out to be a bit complicated as you'll see from looking at that script above, and it turns out there aren't very many cases where `LD_LIBRARY_PATH` ends up getting used (which is why I suggested ignoring it in `search_environment_for_lib()` above).

It seems to me that it would be less likely that you'd run `delocate_xxx()` from within the folder you want to fix libraries in (i.e. you would normally run `delocate_wheel.py <path>/<to>/<wheel>` and not `delocate_wheel.py ./<wheel>`). That means you'd basically never use `LD_LIBRARY_PATH`, as the path(s) to the libraries being delocated will contain a folder in it / them. Why don't I just get rid of that as an option and only support `DYLD_` paths? It will keep everything simple, and searching for libraries will be the same everywhere then (unlike now, where `LD_LIBRARY_PATH` is used when resolving `@rpath` entries and nowhere else).

Reply:
Right, I've stripped out all the LD_LIBRARY_PATH use -- things should be a lot more consistent now.